firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/SharedPreferencesUtil.java

```diff
@@ -190,6 +196,24 @@ public synchronized void clearModelDetails(@NonNull String modelName, boolean cl
         .commit();
   }
 
+  public synchronized Set<CustomModel> listDownloadedModels() {
+    Set<CustomModel> customModels = new HashSet<>();
+    Set<String> keySet = getSharedPreferences().getAll().keySet();
```

Comment:
Since you're reading all the keys at once and not dealing with the sharedpreferences anymore, it may not be necessary to sync the whole method.

Reply:
The second part (added todo) will need to coordinate with android download manager, so I'll need the sync when I add that.